pulsar-zookeeper-utils/src/main/java/com/yahoo/pulsar/zookeeper/ZooKeeperDataCache.java

```diff
@@ -46,11 +47,32 @@ public ZooKeeperDataCache(final ZooKeeperCache cache) {
         this.cache = cache;
     }
 
-    public T get(final String path) throws Exception {
-        return getWithStat(path).getKey();
+    public CompletableFuture<Optional<T>> getAsync(String path) {
+        CompletableFuture<Optional<T>> future = new CompletableFuture<>();
+        cache.getDataAsync(path, this, this).thenAccept(entry -> {
+            future.complete(entry.map(Entry::getKey));
```

Comment:
I think there is a change of NPE:
[ZookeeperCache.getDataAsync](https://github.com/merlimat/pulsar/blob/5b29a8583bfb6608887529105581e2f71b6ae958/pulsar-zookeeper-utils/src/main/java/com/yahoo/pulsar/zookeeper/ZooKeeperCache.java#L257) returns ```null``` as future-result in case of ```NoNodeException```